bgpd/bgp_attr_evpn.c

```diff
@@ -36,6 +36,13 @@
 #include "bgpd/bgp_evpn.h"
 #include "bgpd/bgp_evpn_private.h"
 
+bool bgp_route_evpn_cmp(const struct bgp_route_evpn *e1,
```

Comment:
You're right that we can't compare `struct bgp_route_evnp` using simple memcmp, but this implementation also seems wrong:
1. Why don't you compare `eth_s_id` field?
2. Why you compare `gw_ip` using simple memcmp? If it's IPv4, then we only need to compare 4 bytes and following bytes may be different/uninitialized.

Reply:
Fixed.